config/BUILD.bazel

```diff
@@ -21,14 +21,15 @@ alias(
 filegroup(
     name = "prowjobs",
     srcs = glob([
-        "jobs/**/*.yaml",
+        "jobs/**",
```

Comment:
Will this unintentionally pick up things like markdown and owners files?

Reply:
Yes.

However, if a markdown or owners files happens to also be parseable YAML, then [Prow will unmarshall it](https://github.com/kubernetes/test-infra/blob/b5562fcd6e375d553e267060abef265843615e1c/prow/config/config.go#L726).

Reply:
I'm not certain if that behavior is great, or if it's a bug with Prow.

Reply:
OWNERS files *are* yaml. So, we should probably fix this.

Reply:
> However, if a markdown or owners files happens to also be parseable YAML, then [Prow will unmarshall it](https://github.com/kubernetes/test-infra/blob/b5562fcd6e375d553e267060abef265843615e1c/prow/config/config.go#L726).

That doesn't seem to be the case? Prow only tries to read files with the `.yml` or `.yaml` extensions (which excludes `OWNERS` files and `.md` files). https://github.com/kubernetes/test-infra/blob/b5562fcd6e375d553e267060abef265843615e1c/prow/config/config.go#L696-L698

Reply:
You're right, I'm conflating Prow Config and Job Config.

Prow's Config is loaded in `loadConfig` directly, without checking its extension. This seems fine, since the user points to exactly one file. [src](https://github.com/kubernetes/test-infra/blob/b5562fcd6e375d553e267060abef265843615e1c/prow/config/config.go#L737)

Prow's Job's Config is loaded with `ReadJobConfig`, which checks for `.yaml' and '.yml' extensions

I'll fix this so that it only pulls those two extensions